rotkehlchen/chain/evm/transactions.py

```diff
@@ -456,19 +457,45 @@ def get_or_query_transaction_receipt(self, tx_hash: EVMTxHash) -> 'EvmTxReceipt'
             )
 
         if len(result) == 0:
-            transaction = self.evm_inquirer.get_transaction_by_hash(tx_hash)
-            with self.database.user_write() as write_cursor:
-                dbevmtx.add_evm_transactions(write_cursor, [transaction], relevant_address=None)
-            period = TimestampOrBlockRange(
-                range_type='blocks',
-                from_value=transaction.block_number,
-                to_value=transaction.block_number,
-            )
-            if transaction.to_address is not None:  # internal transactions only through contracts
-                self._query_and_save_internal_transactions_for_range(
-                    address=None,  # get all internal transactions for the parent hash
-                    period=period,
+            if tx_hash == GENESIS_HASH:
+                # Query genesis transactions
+                with self.database.conn.read_ctx() as cursor:
+                    accounts_data = self.database.get_blockchain_account_data(
+                        cursor,
+                        self.evm_inquirer.chain_id.to_blockchain(),
+                    )
+                for data in accounts_data:
```

Comment:
How often can we come here?

Every time the `tx_hash == GENESIS_HASH` we will enter here, if none of the accounts ever have any transactions in the genesis block.

Shouldn't you also check if the account you are iterating has had the block range queried? Cause if it had, then you should not requery etherscan for it every time.